Mage.Tests/src/test/java/org/mage/test/player/TestPlayer.java

```diff
@@ -2196,11 +2196,15 @@ public boolean choose(Outcome outcome, Target target, Ability source, Game game,
             List<UUID> usedTargets = new ArrayList<>();
 
 
+            // TODO: Why TargetAnyTarget not handled there?
```

Comment:
The problem is that TargetCreatureOrPlayer doesn't extend TargetPermanentOrPlayer. TargetAnyTarget does.

`TargetCreatureOrPlayer` is used very rarely, just in Firesong and Sunspeaker and maybe some Star Wars cards, if I recall correctly.

I'd recommend refactoring it instead to inherit sensibly. That should maybe be a separate PR, and then rebase this one, unless you think it's clean and low risk change.

Reply:
Oh that makes sense. I'll make a followup ticket for having `TargetCreatureOrPlayer` extend `TargetPermanentOrPlayer`.

The changes made here are quite safe, no other test was choosing a TargetCreatureOrPlayer, those tests would not have worked. Here I did the same as for TargetPermanentOrPlayer allowing to choose a permanent from the choice pool. (and player can not be choosen, which is also the same behavior)